plugins/file/plugin.go

```diff
@@ -77,7 +77,34 @@ func Handler6(req, resp dhcpv6.DHCPv6) (dhcpv6.DHCPv6, bool) {
 		return nil, false
 	}
 	log.Printf("Found IP address %s for MAC %s", ipaddr, mac)
-	// TODO add an OptIANA based on the above data
+	resp.AddOption(&dhcpv6.OptIANA{
+		// FIXME this must be unique per client address
+		IaId: [4]byte{0xaa, 0xbb, 0xcc, 0xdd},
+		Options: []dhcpv6.Option{
+			&dhcpv6.OptIAAddress{
+				IPv6Addr:          ipaddr,
+				PreferredLifetime: 3600,
+				ValidLifetime:     3600,
+			},
+		},
+	})
+	resp.AddOption(&dhcpv6.OptDNSRecursiveNameServer{
+		NameServers: []net.IP{
+			// FIXME this must be read from the config file
+			net.ParseIP("2001:4860:4860::8888"),
+			net.ParseIP("2001:4860:4860::4444"),
+		},
+	})
+	if oro := req.GetOption(dhcpv6.OptionORO); len(oro) > 0 {
+		for _, code := range oro[0].(*dhcpv6.OptRequestedOption).RequestedOptions() {
```

Comment:
Any chance you'd ever see multiple OROs in one request?

Reply:
According to RFC3315, it should not[1]
```
   Unless otherwise noted, each option may appear only in the options
   area of a DHCP message and may appear only once.  If an option does
   appear multiple times, each instance is considered separate and the
   data areas of the options MUST NOT be concatenated or otherwise
   combined.
```
ORO does not explicitly state that it can appear multiple times[2], so it's only allowed once. I think the correct behaviour is to ignore any occurrence other than the first one, in case of a malformed packet

[1] https://tools.ietf.org/html/rfc3315#section-22
[2] https://tools.ietf.org/html/rfc3315#section-22.7

Reply:
SGTM.